platform/x11/joypad_linux.cpp

```diff
@@ -528,7 +528,7 @@ void JoypadLinux::process_joypads() {
 				input->joy_axis(i, index, joy->curr_axis[index]);
 			}
 		}
-		if (len == 0 || (len < 0 && errno != EAGAIN)) {
+		if (len < 0 && errno != EAGAIN) {
```

Comment:
Are you sure that `len` can't be 0? I didn't check the documentation for `read` but it's not clear to me from the code that it couldn't be used default initialized to `0` if the `while` condition is never true.

Reply:
I just checked `man 3 read` and it says:
```
Upon  successful  completion,  these functions shall return a non-negative integer indicating the number of bytes actually read. Otherwise, the functions shall return −1 and set errno to indicate the error.
```
also:
```
     When attempting to read from an empty pipe or FIFO:

        *  If no process has the pipe open for writing, read() shall return 0 to indicate end-of-file.
```

So as I understand it, we can definitely end up with a value of 0 when reaching EOF, so this check should be kept.

What was the report from the static analyzer?

Reply:
This is the only blocker for merging this PR IMO, so might be worth leave as is for now and reconsider in another issue/PR.